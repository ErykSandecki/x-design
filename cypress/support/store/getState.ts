// types
import { TMainState } from '../../../src/types/reducers';

Cypress.Commands.add('getState', (reducerKey: keyof TMainState, path: any) =>
  cy
    .window()
    .its('store')
    .invoke('getState')
    .then((state: TMainState) => path.split('.').reduce((acc, key) => acc[key], state[reducerKey])),
);
