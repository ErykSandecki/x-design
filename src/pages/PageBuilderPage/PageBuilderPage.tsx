import { FC, useState } from 'react';
import { useSelector } from 'react-redux';

// components
import PanelProperties from 'components/PageBuilder/PanelProperties/PanelProperties';
import Toolbar from 'components/PageBuilder/Toolbar/Toolbar';
import ViewBox from 'components/PageBuilder/ViewBox/ViewBox';

// core
import { RefsProvider } from './core/RefsProvider';

// hooks
import { useKeyDown } from './hooks/useKeyDown';
import { useWheelEvent } from './hooks/useWheelEvent';

// others
import { Box } from 'shared';

// store
import { areaCoordinatesSelector } from 'store/pageBuilder/selectors';

// types
import { MouseMode } from 'types/enums/mouseMode';

const PageBuilderPage: FC = () => {
  const areaCoordinates = useSelector(areaCoordinatesSelector);
  const [coordinates, setCoordinates] = useState(areaCoordinates);
  const [mouseMode, setMouseMode] = useState(MouseMode.default);

  useKeyDown(setMouseMode);
  useWheelEvent();

  return (
    <RefsProvider>
      <Box>
        <Toolbar mouseMode={mouseMode} setMouseMode={setMouseMode} />
        <PanelProperties />
        <ViewBox
          coordinates={coordinates}
          mouseMode={mouseMode}
          setCoordinates={setCoordinates}
          setMouseMode={setMouseMode}
        />
      </Box>
    </RefsProvider>
  );
};

export default PageBuilderPage;
