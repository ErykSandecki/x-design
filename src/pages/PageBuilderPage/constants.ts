// types
import { MouseMode } from 'types';

export const MOUSE_MODE_DISABLED = [MouseMode.comment];
