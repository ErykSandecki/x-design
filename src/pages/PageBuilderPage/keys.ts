// others
import { CONTROL } from 'constant/constants';

// types
import { KeyboardKeys, TKeyboardShortcutsGroup } from 'types';

export const KEYBOARD_SHORTCUTS: TKeyboardShortcutsGroup = {
  alignHorizontalCenter: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.h }],
  alignHorizontalLeft: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.a }],
  alignHorizontalRight: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.d }],
  alignVerticalBottom: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.s }],
  alignVerticalCenter: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.v }],
  alignVerticalTop: [{ primaryKeys: ['alt'], secondaryKey: KeyboardKeys.w }],
  flipHorizontal: [{ primaryKeys: ['shift'], secondaryKey: KeyboardKeys.h }],
  flipVertical: [{ primaryKeys: ['shift'], secondaryKey: KeyboardKeys.v }],
  historyRedo: [{ primaryKeys: [CONTROL, 'shift'], secondaryKey: KeyboardKeys.z }],
  historyUndo: [{ primaryKeys: [CONTROL], secondaryKey: KeyboardKeys.z }],
  mouseModeComment: [{ secondaryKey: KeyboardKeys.e }],
  mouseModeDefault: [{ secondaryKey: KeyboardKeys.escape }, { secondaryKey: KeyboardKeys.q }],
  mouseModeMove: [{ secondaryKey: KeyboardKeys.w }],
  mouseModeToolBeltA: [{ secondaryKey: KeyboardKeys.f }],
  resizeToFit: [{ primaryKeys: ['alt', CONTROL, 'shift'], secondaryKey: KeyboardKeys.r }],
  toggleAutoLayout: [{ primaryKeys: ['shift'], secondaryKey: KeyboardKeys.a }],
};
