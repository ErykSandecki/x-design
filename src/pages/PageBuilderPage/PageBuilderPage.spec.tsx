import { Provider } from 'react-redux';

// components
import PageBuilderPage from './PageBuilderPage';

// mocks
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// utils
import { customRender } from 'test';

describe('PageBuilderPage snapshots', () => {
  it('should render PageBuilderPage', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = customRender(
      <Provider store={store}>
        <PageBuilderPage />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
