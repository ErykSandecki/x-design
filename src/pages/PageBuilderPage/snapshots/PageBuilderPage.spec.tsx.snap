// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBuilderPage snapshots should render PageBuilderPage 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp"
    data-test-box=""
  >
    <div
      class="css-23f94v Toolbar Toolbar--light"
      data-test-box=""
    >
      <div
        class="css-1douohc MouseModes MouseModes--light"
        data-test-box="toolbar"
      >
        <div
          active=""
          class="css-1uzmw43 MouseModes__button MouseModes__button--light MouseModes__button--active MouseModes__button--active--light"
          data-test-box="default"
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="default"
            height="16"
            name="MoveTool"
            width="16"
          />
        </div>
        <div
          class="css-1uzmw43 MouseModes__button MouseModes__button--light"
          data-test-box="toolBeltA"
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="toolBeltA"
            height="16"
            name="FrameTool"
            width="16"
          />
        </div>
        <div
          class="css-1uzmw43 MouseModes__button MouseModes__button--light"
          data-test-box="move"
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="move"
            height="16"
            name="MoveHand"
            width="16"
          />
        </div>
        <div
          class="css-1uzmw43 MouseModes__button MouseModes__button--light"
          data-test-box="comment"
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="comment"
            height="16"
            name="Comment"
            width="16"
          />
        </div>
      </div>
    </div>
    <div
      class="css-k0rgoo PanelProperties PanelProperties--light"
      data-test-box="panel-properties"
      id="panelProperties"
      style="width: 240px;"
    >
      <div
        class="css-xmuf4s PanelProperties__areaHandleResize PanelProperties__areaHandleResize--light"
        data-test-box=""
        data-test-resize=""
      />
      <div
        class="css-1jnxvbp Section Section--light"
        data-test-section=""
      >
        <div
          class="css-kp4ill"
          data-test-box=""
        >
          <small
            class="css-1jnxvbp Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Mode
          </small>
          <div
            class="css-1f1atqu"
            data-test-box=""
          />
        </div>
        <div
          class="css-gu5024 Tabs Tabs--light"
          data-test-tabs="panel-mode"
        >
          <div
            active=""
            class="css-17z9djj Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
            data-test-tab="design"
          >
            Design
          </div>
          <div
            class="css-17z9djj Tabs__tab Tabs__tab--light"
            data-test-tab="prototype"
          >
            Prototype
          </div>
        </div>
      </div>
      <div
        class="PanelProperties__sections PanelProperties__sections--light"
      >
        <div
          class="css-1jnxvbp Section Section--light"
          data-test-section="background"
        >
          <div
            class="css-kp4ill"
            data-test-box=""
          >
            <small
              class="css-1jnxvbp Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Page
            </small>
            <div
              class="css-1f1atqu"
              data-test-box=""
            />
          </div>
          <div
            class="css-1jnxvbp SectionColumn SectionColumn--light"
            data-test-section=""
          >
            <div
              class="css-xdzdnq SectionColumnLabels SectionColumnLabels--light"
              data-test-box=""
              style="width: calc(100% - 24px - 0px - 8px);"
            />
            <div
              class="css-4ja0ji"
              data-test-box=""
            >
              <div
                class="css-biptg5 SectionColumnContent SectionColumnContent--light SectionColumnContent--single SectionColumnContent--single--light"
                data-test-box=""
                style="width: calc(100% - 24px - 0px - 8px);"
              >
                <div
                  class="css-1jnxvbp FieldGroup FieldGroup--light"
                  data-test-field-group="background"
                >
                  <div
                    class="css-1jnxvbp ColorPicker__colorInput ColorPicker__colorInput--light TextField TextField--light"
                    data-test-text-field="color"
                    data-test-tooltip=""
                  >
                    <div
                      aria-describedby="test-id"
                      class="css-1jnxvbp"
                      data-test-color="background"
                    >
                      <div
                        class="css-1axzf39 Color Color--light"
                        data-test-box=""
                      >
                        <div
                          class="css-agg4zg Color__picker Color__picker--light"
                          data-test-box=""
                          style="background-color: rgb(30, 38, 47);"
                        />
                        <div
                          class="css-agg4zg Color__pickerAlpha Color__pickerAlpha--light"
                          data-test-box=""
                          style="background-color: rgb(30, 38, 47);"
                        />
                        <div
                          class="css-hnxq09 Color__pickerTexture Color__pickerTexture--light"
                          data-test-box=""
                        />
                      </div>
                    </div>
                    <input
                      class="TextField__input TextField__input--light"
                      data-test-text-field-input="color"
                      maxlength="6"
                      value="1e262f"
                    />
                  </div>
                  <div
                    class="css-1jnxvbp ColorPicker__alphaInput ColorPicker__alphaInput--light TextField TextField--light"
                    data-test-text-field="alpha"
                    data-test-tooltip=""
                  >
                    <input
                      class="TextField__input TextField__input--light"
                      data-test-text-field-input="alpha"
                      max="100"
                      maxlength="6"
                      min="0"
                      type="number"
                      value="100"
                    />
                    <div
                      class="css-e5vky9 ScrubbableInput ScrubbableInput--light"
                      data-test-box=""
                      data-test-scrubbable-input=""
                    >
                      <div
                        class="ColorPicker__alphaInputUnit ColorPicker__alphaInputUnit--light"
                      >
                        %
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-1jnxvbp"
                data-test-box=""
              >
                <div
                  class="css-c1qms7"
                  data-test-box=""
                >
                  <div
                    class="css-1jnxvbp ButtonIcon ButtonIcon--light"
                    data-test-button-icon="toggle-visibility"
                    data-test-tooltip=""
                  >
                    <div
                      class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                      data-test-icon="eyes-opened"
                      height="14"
                      name="EyesOpened"
                      width="14"
                    />
                  </div>
                  <div
                    class="Tooltip Tooltip--light"
                  >
                    <div
                      class="Tooltip__content Tooltip__content--light"
                    >
                      Toggle visibility
                      <div
                        class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                        data-test-icon="carrot-down"
                        height="14"
                        name="CarrotDown"
                        width="10"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-117yijb ViewBox ViewBox--light ZoomBox ZoomBox--light ZoomBox--idle ZoomBox--idle--light ZoomBox--default ZoomBox--default--light"
      data-test-box="zoom-box"
      tabindex="0"
    >
      <div
        class="css-1jnxvbp ZoomBox__backgroundMask ZoomBox__backgroundMask--light"
        data-test-box="zoom-box-background-mask"
        style="background-color: rgb(30, 38, 47); display: initial;"
      />
      <div
        class="css-1jnxvbp ZoomBox__textureBlank ZoomBox__textureBlank--light"
        data-test-box=""
        style="transform: translate(0px, 0px) scale(1);"
      />
      <div
        class="css-1j4c14e ZoomBox__zoomContent ZoomBox__zoomContent--light"
        data-test-box=""
        id="zoomContentId"
        style="transform: translate(0px, 0px) scale(1);"
      >
        <div
          class="css-1jnxvbp Elements__element Elements__element--light Frame Frame--light Element Element--light"
          data-status="false"
          data-test-element="mh4j5eb01761290939051"
          id="mh4j5eb01761290939051"
          style="align-items: flex-start; display: flex; flex-direction: column; justify-content: flex-start; row-gap: 0px; left: 637px; top: 323px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; background-color: rgb(255, 255, 255); height: 328px; position: absolute; width: 382px;"
        >
          <div
            class="css-1jnxvbp Frame__wrapper Frame__wrapper--light"
            data-test-box=""
            style="height: 328px; left: 637px; top: 323px; transform: rotate(0deg); width: 382px;"
          >
            <small
              class="css-1jnxvbp Frame__label Frame__label--light Frame__label--top Frame__label--top--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Frame
            </small>
          </div>
        </div>
        <div
          class="css-u5h2b"
          data-test-box=""
          id="overlayContainer"
          style="z-index: var(--xd-page-builder-zindex-overlay-container);"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
